google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/data/v2/stub/metrics/HeaderTracerCallableTest.java

```diff
@@ -0,0 +1,461 @@
+        .when(fakeService)
+        .readRows(any(ReadRowsRequest.class), anyObserver(ReadRowsResponse.class));
+    doAnswer(new MutateRowAnswer())
+        .when(fakeService)
+        .mutateRow(any(MutateRowRequest.class), anyObserver(MutateRowResponse.class));
+
+    // Make a few calls to the server which will inject the server-timing header and the counter
+    // should be 0.
+    stub.readRowsCallable().call(Query.create(TABLE_ID));
+    stub.mutateRowCallable().call(RowMutation.create(TABLE_ID, "key"));
+
+    Thread.sleep(WAIT_FOR_METRICS_TIME_MS);
+    long mutateRowMissingCount =
+        StatsTestUtils.getAggregationValueAsLong(
+            localStats,
+            RpcViewConstants.BIGTABLE_GFE_HEADER_MISSING_COUNT_VIEW,
+            ImmutableMap.of(RpcMeasureConstants.BIGTABLE_OP, TagValue.create("Bigtable.MutateRow")),
+            PROJECT_ID,
+            INSTANCE_ID,
+            APP_PROFILE_ID);
+    long readRowsMissingCount =
+        StatsTestUtils.getAggregationValueAsLong(
+            localStats,
+            RpcViewConstants.BIGTABLE_GFE_HEADER_MISSING_COUNT_VIEW,
+            ImmutableMap.<TagKey, TagValue>of(
+                RpcMeasureConstants.BIGTABLE_OP, TagValue.create("Bigtable.ReadRows")),
+            PROJECT_ID,
+            INSTANCE_ID,
+            APP_PROFILE_ID);
+
+    Thread.sleep(WAIT_FOR_METRICS_TIME_MS);
+
+    assertThat(mutateRowMissingCount).isEqualTo(0);
+    assertThat(readRowsMissingCount).isEqualTo(0);
+
+    // Make a few more calls to the server which won't add the header and the counter should match
+    // the number of requests sent.
+    int readRowsCalls = new Random().nextInt(10) + 1;
+    int mutateRowCalls = new Random().nextInt(10) + 1;
+    for (int i = 0; i < mutateRowCalls; i++) {
+      noHeaderStub.mutateRowCallable().call(RowMutation.create(TABLE_ID, "fake-key" + i));
+    }
+    for (int i = 0; i < readRowsCalls; i++) {
+      noHeaderStub.readRowsCallable().call(Query.create(TABLE_ID));
+    }
+
+    Thread.sleep(WAIT_FOR_METRICS_TIME_MS);
+
+    mutateRowMissingCount =
+        StatsTestUtils.getAggregationValueAsLong(
+            localStats,
+            RpcViewConstants.BIGTABLE_GFE_HEADER_MISSING_COUNT_VIEW,
+            ImmutableMap.of(RpcMeasureConstants.BIGTABLE_OP, TagValue.create("Bigtable.MutateRow")),
+            PROJECT_ID,
+            INSTANCE_ID,
+            APP_PROFILE_ID);
+    readRowsMissingCount =
+        StatsTestUtils.getAggregationValueAsLong(
+            localStats,
+            RpcViewConstants.BIGTABLE_GFE_HEADER_MISSING_COUNT_VIEW,
+            ImmutableMap.<TagKey, TagValue>of(
+                RpcMeasureConstants.BIGTABLE_OP, TagValue.create("Bigtable.ReadRows")),
+            PROJECT_ID,
+            INSTANCE_ID,
+            APP_PROFILE_ID);
+
+    assertThat(mutateRowMissingCount).isEqualTo(mutateRowCalls);
+    assertThat(readRowsMissingCount).isEqualTo(readRowsCalls);
+  }
+
+  private class ReadRowAnswer implements Answer {
```

Comment:
I think this is a personal taste thing, but why go through mockito for this? Wouldnt it be simpler to just implement a fake BigtableImplBase?